### PR TITLE
Create article form

### PR DESCRIPTION
**Pull request by arnaud-blanc**

### CODE REVIEW

**Review by Orijma — APPROVED** [review]

**Review by Orijma — DISMISSED** [review]

File: README.md [hunk 1]
```diff
@@ -31,7 +31,16 @@ It's symfony website-skeleton project with some additional tools to validate cod
...
 3. Run `yarn install`
-4. Run `yarn encore dev` to build assets
+4. Run `yarn encore dev` to build assets  
+*Option :*
+5. If you encounter this error message : <u>"Only the ‘en’ locale is supported. Please install the ‘intl’ extension for full localization capabilities.</u>"  
+Please read [Twig > Docs > Filters > format_datetime](https://twig.symfony.com/doc/3.x/filters/format_datetime.html)  
+5.0. Run `composer recipes` to check if *twig/intl-extra, twig/extra-bundle* are available, if not :  
+5.1. Run `composer require [twig/intl-extra] [twig/extra-bundle]`  
+5.2. Install php7.4-intl module : `sudo apt install php7.4-intl` (If you run php7.4...)  
+5.3. Restart apache : `service apache2 restart` (Is it really necessary?)
```
Comment (RomainClair) [inline README.md:41]: Depends if you run apache or not...
Do you really need the intl component as you don't have i18n in your project for now ?

File: src/Form/ArticleType.php [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+                'required' => false,
+                'constraints' => [
+                    new File([
+                        'maxSize' => '1024k',
+                        'mimeTypes' => [
+                            'image/gif',
+                            'image/jpg',
+                            'image/jpeg',
+                            'image/png',
+                        ],
```
Comment (RomainClair) [inline src/Form/ArticleType.php:35]: There is a Image constraint for image File

**Review by RomainClair — CHANGES_REQUESTED** [review]
.gitignore is not retroactive.
You have to remove the unwanted file from the repository.

File: README.md (hunk 1 above)
Comment (arnaud-blanc) [reply README.md:41]: The `intl` component is needed to give a local format to the dates that will be displayed in the views.
Please read :
https://twig.symfony.com/doc/3.x/filters/format_datetime.html

File: src/Form/ArticleType.php (hunk 2 above)
Comment (arnaud-blanc) [reply src/Form/ArticleType.php]: Constraint has been updated with Image.

**Review by Orijma — DISMISSED** [review]

File: README.md (hunk 1 above)
Comment (RomainClair) [reply README.md:41]: But now, it's in the composer.json/lock so the composer install command will install the intl-extra

File: src/Controller/ArticleController.php [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+                    throw $e;
+                }
+                $article->setImageFilename($newFilename);
+            }
+            $entityManager->persist($article);
+            $entityManager->flush();
+
+            $this->addFlash('success', "L'article a bien été créé.");
+
+            return $this->redirectToRoute($articleRequest['article']['type']);
```
Comment (RomainClair) [inline src/Controller/ArticleController.php:69]: route "actuelles" does not exists

File: src/Entity/Article.php [hunk 4]
```diff
@@ -3,13 +3,20 @@
...
 use App\Repository\ArticleRepository;
+use DateTime;
 use Doctrine\ORM\Mapping as ORM;
+use Symfony\Component\Validator\Constraints as Assert;
+use Symfony\Bridge\Doctrine\Validator\Constraints\UniqueEntity;
 
 /**
  * @ORM\Entity(repositoryClass=ArticleRepository::class)
+ * @UniqueEntity("title")
+ * @UniqueEntity("content")
```
Comment (RomainClair) [inline src/Entity/Article.php:14]: Is it really releveant to have unique content ?

File: src/Entity/Article.php [hunk 5]
```diff
@@ -19,28 +26,39 @@ class Article
 
     /**
      * @ORM\Column(type="string", length=255)
```
Comment (RomainClair) [inline src/Entity/Article.php:28]: Unique fields should probably also have the unique ORM annotation

**Pull request closed without merging**